ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -75,6 +93,35 @@ private[apiserver] final class RateLimitingInterceptor(metrics: Metrics, config:
     }
   }
 
+  private def memoryOverloaded(fullMethodName: String): Either[String, Unit] = {
+    tenuredMemoryPool.fold[Either[String, Unit]](Right(())) { p =>
+      if (p.isCollectionUsageThresholdExceeded) {
+        // Based on a combination of JvmMetricSet and MemoryUsageGaugeSet
+        val poolBeanMetricPrefix = s"jvm_memory_usage_pools_${p.getName}"
+        val rpcStatus =
+          s"""
+             | The ${p.getName} collection  has exceeded the maximum (${p.getCollectionUsageThreshold}).
+             | The rejected call was $fullMethodName.
+             | Jvm memory metrics are available at $poolBeanMetricPrefix
+          }.
+          """.stripMargin
+        gc()
+        Left[String, Unit](rpcStatus)
+      } else {
+        Right(())
+      }
+    }
+  }
+
+  /** When the collected tenured memory pool usage exceeds the threshold this state will continue even if memory
+    * has been freed up if not garbage collection takes place.  For this reason when we are over limit we also
+    * run garbage collection on every request to ensure the collection usage stats are as up to date as possible
+    * to thus stop rate limiting as soon as possible.
+    */
+  private def gc(): Unit = {
+    memoryMxBean.gc()
```

Comment:
As it basically delegates to `System.gc()` as according to `MemoryMXBean.java` I would use directly here `System.gc()` as then it clear what is done.

Checking the necessity of calling explicitly a garbage collector deserves a deep analysis on my side. There are usually good reasons exist not to ever call it explicitly :)